packages/block-editor/src/store/reducer.js

```diff
@@ -1448,6 +1448,26 @@ export function automaticChangeStatus( state, action ) {
 	// Reset the state by default (for any action not handled).
 }
 
+/**
+ * Reducer returning current highlighted block.
+ *
+ * @param {boolean} state  Current highlighted block.
+ * @param {Object}  action Dispatched action.
+ *
+ * @return {string} Updated state.
+ */
+export function highlightedBlock( state = '', action ) {
+	switch ( action.type ) {
+		case 'HIGHLIGHT_BLOCK':
+			return action.clientId;
+
+		case 'UNHIGHLIGHT_BLOCK':
+			return '';
```

Comment:
I think `null` would be a more meaningful data type to express the "unset" state. There's quite a bit of precedent as well (`block-editor`'s `insertionPoint` state, `editor`'s `postId` and `postType` states, etc).